src/allmydata/uri.py

```diff
@@ -725,12 +744,15 @@ def get_verify_cap(self):
         return None
 
 
-ALLEGED_READONLY_PREFIX = 'ro.'
-ALLEGED_IMMUTABLE_PREFIX = 'imm.'
+ALLEGED_READONLY_PREFIX = b'ro.'
+ALLEGED_IMMUTABLE_PREFIX = b'imm.'
 
 def from_string(u, deep_immutable=False, name=u"<unknown name>"):
-    if not isinstance(u, str):
-        raise TypeError("URI must be str: %r" % (u,))
+    """Create URI from either unicode or byte string."""
+    if isinstance(u, unicode):
+        u = u.encode("utf-8")
+    if not isinstance(u, bytes):
+        raise TypeError("URI must be unicode string or bytes: %r" % (u,))
```

Comment:
Is there anything interesting or worth documenting in this change?  I can't think of anything from eyeballing the tests or running this in a REPL, so I thought I should just ask.

Reply:
Don't think so, it's not meaningful insofar as so far everything is bytes anyway. Just added that in case it turns out to be useful for Python 3.